src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/AssignSessionsToHearingAction.java

```diff
@@ -0,0 +1,161 @@
+package uk.gov.hmcts.reform.sandl.snlevents.actions.hearing;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.hmcts.reform.sandl.snlevents.actions.Action;
+import uk.gov.hmcts.reform.sandl.snlevents.actions.interfaces.RulesProcessable;
+import uk.gov.hmcts.reform.sandl.snlevents.exceptions.SnlRuntimeException;
+import uk.gov.hmcts.reform.sandl.snlevents.messages.FactMessage;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.Hearing;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.HearingPart;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.Session;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.UserTransactionData;
+import uk.gov.hmcts.reform.sandl.snlevents.model.request.HearingSessionRelationship;
+import uk.gov.hmcts.reform.sandl.snlevents.model.request.SessionAssignmentData;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.HearingRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.SessionRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.service.RulesService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+
+public class AssignSessionsToHearingAction extends Action implements RulesProcessable {
+
+    private final EntityManager entityManager;
+    protected HearingSessionRelationship relationship;
```

Comment:
Is there any reason why these are protected?

Reply:
adjusting to the style of other actions; which means entityManager is off

Reply:
Having a deeper thought, I assume idea was that these Actions can be further extended in a different package where protected will be more useful then default access.
In both cases against 'private', tests can check fields value after certain steps, for example, act() method.

Reply:
I'd keep them protected. In next sprint we're going to have AdjurnHearingAction and VacateHearingAction and the only difference between them is the status name. That means extending and it will be useful to access parent's properties